app/services/decs_snippet_service.rb

```diff
@@ -0,0 +1,11 @@
+class DECSSnippetService
```

Comment:
I'm all for code organization but I think this service is a bit redundant, since you only issues a GET :) I'd just put the `HTTParty.get` inside the liquid tag.

Reply:
Okie I can remove the service for now. But, I can definitely see that this will be extended to enable more features in the near future.

Reply:
we'll get there when we get there I guess, right now I think the code should be self contained.

Now that I think about it, i'm not even sure the liquid tag should be calling DECS everytime is rendered, why not just checking if the URL is valid?

Most of the other liquid tags will just render nothing if the ID given is not actually existent, without incurring in the overhead of checking upstream with an HTTP call.

Reply:
Currently, the main reason was to determine the height of the snippet and also to show appropriate error messages. Are you concerned about the performance overhead with the GET request ??

Reply:
Yes, exactly

Reply:
Is there a way to get the height without sending an HTTP call? I think with other Liquid tags, the height is automatically assigned by the HTML from the embed. Is that not the case with DECS?

Nice job implementing this Liquid tag though!

Reply:
Hi @Zhao-Andy, thanks for looking into this. Currently, it's not the case. I will make the necessary changes on our side to enable auto height adjustments. Should be done by early next week. 

Having said that. I tried using JS magic to set the height of the iframe dynamically but the JS code doesn't seem to be executed upon rendering. Am I missing something there?

Reply:
Hmm, are you having issues with rendering it in `preview`? If so, @maestromac might have some ideas on this.

How are your running the JS code?